src/Microsoft.AspNetCore.Hosting/Internal/ApplicationLifetime.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNetCore.Hosting.Internal
     /// <summary>
     /// Allows consumers to perform cleanup during a graceful shutdown.
     /// </summary>
-    public class ApplicationLifetime : IApplicationLifetime
+    public class ApplicationLifetime : IApplicationLifetime, Extensions.Hosting.IApplicationLifetime
```

Comment:
Isn't this breaking?

Reply:
Why would it be?

Reply:
It's interesting that you choose to have the concrete class implement both interfaces rather than having the interfaces chain. Consumers must explicitly choose one interface or the other. Wouldn't chaining them make the transition even smoother? An app could convert one method at a time and work backwards.

E.g. MyMethod(Ex.H.IApplicationLifetime lifetime) could accept either Extensions.Hosting.IApplicationLifetime or AspNetCore.Hosting.IApplicationLifetime.

Or do you assume everything will come directly from DI and they only need to update one constructor?

Reply:
The goal is to have new code take in extensions hosting where possible via ctor injection. This change also doesn’t preclude the derivation if we see benefits there.